### PR TITLE
ULTRA L1c fix positional uncertainties 

### LINKED ISSUE #3067: ULTRA l2 positional uncertainty is all zero in l2 maps
example: 

this prints 0 and 0

from imap_processing.cdf.utils import load_cdf
map4 = load_cdf("/Users/luco3133/projects/imap_processing/data/imap/ultra/l2/2025/11/imap_ultra_l2_u90-ena-h-sf-nsp-full-hae-6deg-4mo_20251114_v000.cdf")
positional_uncert_theta = map4["positional_uncert_theta"][0,1,:]
#print(positional_uncert_theta.data)
print(max(positional_uncert_theta.data))
print(min(positional_uncert_theta.data)) 

I need to investigate why this is happening and where.

### PR DESCRIPTION
**Pull request by lacoak21**
# Change Summary

closes #3067

## Overview

Only accumulate finite values. This was slowly propagating nans into the summation and the final arrays were all only zero or nan. 

## File changes

- imap_processing/ultra/l1c/l1c_lookup_utils.py
  - fix summation 

## Testing

Add test to ensure that values are not only zero or nan

### CODE REVIEW

File: imap_processing/ultra/l1c/l1c_lookup_utils.py [hunk 1]
```diff
@@ -303,15 +311,19 @@ def calculate_accepted_pixels(  # noqa: PLR0912
...
-            fwhm_theta_sum[:, accepted_pix] += fwhm_theta.T
-            fwhm_phi_sum[:, accepted_pix] += fwhm_phi.T
-            sample_count[:, accepted_pix] += 1
+            # Accumulate FWHM values where theta and phi are finite
+            valid_fwhm = np.isfinite(fwhm_theta) & np.isfinite(
+                fwhm_phi
+            )  # (npix, n_energy)
+            fwhm_theta_sum[:, accepted_pix] += np.where(valid_fwhm, fwhm_theta, 0.0).T
+            fwhm_phi_sum[:, accepted_pix] += np.where(valid_fwhm, fwhm_phi, 0.0).T
+            sample_count[:, accepted_pix] += valid_fwhm.T.astype(sample_count.dtype)
```
Comment (vmartinez-cu) [inline imap_processing/ultra/l1c/l1c_lookup_utils.py:320]: consider adding a comment about why the dimensions are transposed here if it hasn't already been described elsewhere in this file

File: imap_processing/ultra/l1c/l1c_lookup_utils.py [hunk 2]
```diff
@@ -260,10 +260,18 @@ def calculate_accepted_pixels(  # noqa: PLR0912
                 if reject_scattering:
                     valid_pixels[i, e_ind, accepted_pix] = scattering_mask.flatten()
 
-                # Accumulate FWHM values
-                fwhm_theta_sum[e_ind, accepted_pix] += fwhm_theta.flatten()
-                fwhm_phi_sum[e_ind, accepted_pix] += fwhm_phi.flatten()
-                sample_count[e_ind, accepted_pix] += 1
+                # Accumulate FWHM values only where BOTH theta and phi are finite
+                valid_fwhm = np.isfinite(fwhm_theta) & np.isfinite(fwhm_phi)
```
Comment (vmartinez-cu) [inline imap_processing/ultra/l1c/l1c_lookup_utils.py:264]: Are zeros valid? If so, is it possible that the accumulation that happens next could sum to zero? Just curious since your test asserts that theta and phi are greater than zero and this mask doesn't exclude zeros. It may not be an issue though if the accumulation is always greater than zero

**Review by vmartinez-cu — APPROVED** [review]
I just have one question that isn't a blocker. Nice job using a boolean mask to exclude nans in the computations!

File: imap_processing/ultra/l1c/l1c_lookup_utils.py (hunk 2 above)
Comment (lacoak21) [reply imap_processing/ultra/l1c/l1c_lookup_utils.py:264]: Yes! Zeros are valid! The test is set up in a way that none of those values should be zero though.

File: imap_processing/ultra/l1c/l1c_lookup_utils.py (hunk 1 above)
Comment (lacoak21) [reply imap_processing/ultra/l1c/l1c_lookup_utils.py:320]: ah good call!
